bundles/org.openhab.binding.bluetooth.dbusbluez/src/main/java/org/openhab/binding/bluetooth/dbusbluez/handler/DBusBlueZBridgeHandler.java

```diff
@@ -0,0 +1,343 @@
+            return;
+        }
+
+        logger.debug("Creating BlueZ adapter with address '{}'", adapterAddress);
+
+        if (!initializeDeviceManager()) {
+            // Device manager not initialized so exiting.
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, "Library error.");
+            return;
+        }
+
+        if (!initializeAdapter()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "Bluetooth adapter could not be found.");
+            return;
+        }
+
+        // Beyond this point, lib is initialized and adapter was found.
+        updateStatus(ThingStatus.ONLINE);
+
+        this.discoveryJob = scheduler.scheduleAtFixedRate(this::refreshDevices, 10, 10, TimeUnit.SECONDS);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    /**
+     * This method is essential as most cheap USB chips will not supporte scanning
+     * AND connecting simultaneously. The scanning state can also be changed by
+     * the operating system or bluetoothctl command, so it is mandatory to check
+     * state before trying to connect.
+     *
+     * @return
+     */
+    public boolean isCurrentlyScanning() {
+        boolean scanning = this.adapter.isDiscovering();
+        logger.debug("Is adapter currently scanning : {}", scanning);
+        return scanning;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public BluetoothAddress getAddress() {
+        return adapterAddress;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public void dispose() {
+
+        logger.debug("Termination of DBus BlueZ handler");
+
+        if (this.discoveryJob != null) {
+            this.discoveryJob.cancel(true);
+            this.discoveryJob = null;
+        }
+
+        if (this.adapter != null) {
+            this.adapter.stopDiscovery();
+            this.adapter = null;
+        }
+
+        if (this.deviceManager != null) {
+            this.deviceManager.closeConnection();
+            this.deviceManager = null;
+        }
+
+        super.dispose();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    private void startDiscovery() {
+        // we need to make sure the adapter is powered first
+        if (!adapter.isPowered()) {
+            adapter.setPowered(true);
+        }
+        if (Boolean.FALSE.equals(adapter.isDiscovering())) {
+            adapter.startDiscovery();
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+
+    private void refreshDevices() {
+        logger.debug("refreshDevices()");
+
+        if (lockDiscoveryJob.tryLock()) {
```

Comment:
You shouldn't need to do locking like this. If you really wanted to you should synchronize the method or around a lock object. 

```
private final Object discoveryLock = new Object();
...
private void refreshDevices(){
    synchronized(discoveryLock){
...
    }
}

Reply:
The reason why I decided to do this instead of synchronized is I did not want to make it blocking : if there is already a refreshDevice ongoing, it is not important to delay the next call and we can skip one refreshDevice iteration.